test/src/test_main.py

```diff
@@ -0,0 +1,185 @@
+import jsonschema
+import json
+import copy
+import pytest
+
+SCHEMA_PATH = "json-schema/StructuredQuery-schema-v3.json"
+EXAMPLE_JSON_PATH = "example-json/sqv3_all_properties.json"
+
+TERM_CODE = {"code": "somecode-4591", "system": "somesystem-1138", "display": "somedisplay-5832"}
+UNIT = {"code": "somecode-8331", "display": "somedisplay-1269"}
+ATTRIBUTE_CODE = {"code": "somecode-0103", "system": "somesystem-3571", "display": "somedisplay-9567"}
+
+
+@pytest.fixture
+def schema():
+    loaded_schema = None
+    with open(SCHEMA_PATH, 'r') as file:
+        loaded_schema = json.load(file)
+    return loaded_schema
+
+@pytest.fixture
+def basic_sq():
+    sq = {}
+    sq["inclusionCriteria"] = [[ {"termCodes": [TERM_CODE]}]]
+    sq["version"] = "someVersion"
+    return sq
+    
+def test_basic_sq(schema, basic_sq):
+    jsonschema.validate(basic_sq, schema)
+
+def test_example_json(schema):
+    example_json = None
+    with open(EXAMPLE_JSON_PATH, 'r') as file:
+        example_json = json.load(file)
+
+    jsonschema.validate(example_json, schema)
+
+def test_additional_prop_in_criterion(schema, basic_sq):
+    sq = copy.copy(basic_sq)
+    jsonschema.validate(sq, schema)
+
+    sq["inclusionCriteria"][0][0]["foo"] = "bar"
+    with pytest.raises(jsonschema.exceptions.ValidationError):
+        jsonschema.validate(sq, schema)
+
+def test_additional_prop_in_outermost_layer(schema, basic_sq):
+    sq = copy.copy(basic_sq)
+    jsonschema.validate(sq, schema)
+
+    sq["foo"] = "bar"
+    with pytest.raises(jsonschema.exceptions.ValidationError):
+        jsonschema.validate(sq, schema)
+
+def test_additional_prop_in_termcode(schema, basic_sq):
+    sq = copy.deepcopy(basic_sq)
```

Comment:
Why do you need a `deepcopy` here but use a `copy` on the other tests?

Reply:
because in this test i change the the termocode, which in a shallow copy would be a reference to the original termcode. In the other tests, i dont change original references.

Reply:
Ok then I would use `deepcopy` everywhere, because it is a shared data structure and the debugging will be expensive if someone changes something in one test and the error occurs in another test.

Reply:
Another solution would be to not use a fixture and instead call the `basic_sq` function every time you need one.